pkg/component/devfile/adapters/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,69 @@
+package kubernetes
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/component/devfile/adapters/common"
+	devfileCommon "github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Adapter maps Devfiles to Kubernetes resources and actions
+type Adapter struct {
+	common.DevfileAdapter
+	Client kclient.Client
+}
+
+// Start creates Kubernetes resources that correspond to the devfile if they don't already exist
+func (k Adapter) Start() error {
+
+	err := k.Devfile.Data.Validate()
+	if err != nil {
+		return errors.Wrap(err, "The devfile is invalid")
+	}
+
+	componentName := k.DevfileComp.Name
+
+	if k.componentExists(componentName) {
```

Comment:
wouldn't be possible that some env variables or resource Requirements have changed? shouldn't we verify if the local state is same as the remote state?

Reply:
Yes, but I scoped these changes to handle creation only and to get the very basic scenarios working. That is, use the push command for creation and the delete command for deletion. Handling update correctly is more involved because the list of components in the devfile could change as well and we cannot currently dynamically add/remove containers from pods without restarting the entire pod. I think we need to discuss what our initial and long term update strategy should be. I have added a topic to the agenda to discuss it on the odo contributors meeting.

Reply:
On the call today, @kadel mentioned we can just create the deployment spec according to the current devfile and if a deployment already exists then issue an update instead of a create and that should work. I will try it out and apply the changes to the branch.

Reply:
That approach works well, I've updated the PR